### PR TITLE
Implement min aggregation

### PR DESCRIPTION
**Pull request by guilload**
Just needed the `min` for something I'm working on in Quickwit. Will add `max` if this change looks ok.

### CODE REVIEW

**Comment by adamreichold** [comment]
Is there a measurable performance benefit from using this versus `Stats` and ignoring everything but the `min` field? If there is, could we deduplicate the implementation by adding a single aggregation using any associative function `Fn(T, T) -> T`?

**Comment by PSeitz** [comment]
> Is there a measurable performance benefit from using this versus `Stats` and ignoring everything but the `min` field? If there is, could we deduplicate the implementation by adding a single aggregation using any associative function `Fn(T, T) -> T`?

Yes, that was also my first thought. There's a stats benchmark already, it should be easy to compare with `bench_aggregation_stats_f64`

**Comment by fulmicoton** [comment]
There is a min aggregation in elastic. (apart from stat) 
That's enough reason to have one. :)

We could reuse the implementation though, if computing just the min is not faster. 
Is this how average is done today?

File: src/aggregation/metric/min.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    pub fn field_name(&self) -> &str {
+        &self.field
+    }
+}
+
+/// Holds the intermediate state of the minimum computation.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct IntermediateMin {
+    pub(crate) min: f64,
+    pub(crate) is_some: bool,
```
Comment (fulmicoton) [inline src/aggregation/metric/min.rs:45]: `Option<f64>` ?

**Comment by PSeitz** [comment]
> There is a min aggregation in elastic. (apart from stat) That's enough reason to have one. :)
> 
> We could reuse the implementation though, if computing just the min is not faster. Is this how average is done today?


Yes, just code reuse, that's what I meant. Average has its own code, since it was there first. It could be refactored though.
The bench looks pretty much the same

```
test aggregation::tests::bench::bench_aggregation_average_f64                                                            ... bench:   8,574,992 ns/iter (+/- 283,628)
test aggregation::tests::bench::bench_aggregation_stats_f64                                                              ... bench:   8,235,065 ns/iter (+/- 319,069)
```

**Comment by fulmicoton** [comment]
@PSeitz let's put the code in common then

**Pull request closed without merging**